File: Haskell/Atividade IV/problema15.hs
-- func :: Int -> Int
-- func r =


mdc :: Int -> Int -> Int
mdc _x 0 = _x
mdc x y =
	mdc y (mod x y)

main = do
	ns <- getLine
	let n = (read ns :: Int)
	let r = 1
	let i = r
	let p = 0
	let j = p

	if r < n then
		if (mdc n r == 1) then
			p <- p + 1
			r <- r + 1
	else
		print(p)

File: PROLOG/Atividade XV/problema15.pl
mdc(X,Y,Resultado) :-
    Resto is X mod Y,
    (Resto == 0 -> Resultado is Y; mdc(Y,Resto,Resultado)).
% ?- mdc(36,5,R).
% R = 1.

% ?- mdc(36,6,R).
% R = 6.

% ?- mdc(348,156,R).
% R = 12.